main.go

```diff
@@ -18,18 +39,69 @@ func getPing(w http.ResponseWriter, r *http.Request) {
 
 }
 
-func getBooks(w http.ResponseWriter, r *http.Request) {
+func Books(w http.ResponseWriter, r *http.Request) {
+	metodo := r.Method
+	fmt.Printf("metodo: %v\n", metodo)
+	switch {
+	case metodo == "GET":
+		getBooks(w, r)
+	case metodo == "POST":
+		postBooks(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
```

Comment:
1 - Avoid using string literals in comparisons, use the `http` package constants instead.
2 - Instead of a naked switch `switch {`, you can use method as the `switch` target and simplify the syntax. 
3 - You want to return after the dedicated http method function calls. Otherwise (if you had any code after) you would continue execution and it could cause issues. Specially after some time, when you want to tweak this code a little and you already forgot how it actually works.
```suggestion
	switch method {
	case http.MethodGet:
		return getBooks(w, r)
	case http.MethodPost:
		return postBooks(w, r)
	default:
		return w.WriteHeader(http.StatusMethodNotAllowed)
	}

Reply:
VScode did not allow me to return a function:
- getBooks(w, r) (no value) used as value
- postBooks(w, r) (no value) used as value
- w.WriteHeader(http.StatusMethodNotAllowed) (no value) used as value

But this way it worked:
switch method {
	case http.MethodGet:
		getBooks(w, r)
		return
	case http.MethodPost:
		postBooks(w, r)
		return
	default:
		w.WriteHeader(http.StatusMethodNotAllowed)
		return
	}

Reply:
That's perfect. It was my mistake. Sorry.